### ISSUE TITLE
Allow custom reconnect behaviour

### ISSUE THREAD

**Issue by Anntoin** [robtaussig/react-use-websocket#110]
I like to be able to customise the retry behaviour, e.g. like using a token bucket. Then a client would be able to reconnect immediately for 'normal' websocket connection interruptions but with a persistent error or connection issue the client appropriately throttles it's requests.

I'm thinking this could be accomplished by allowing `reconnectInterval` to optionally take a function.

Is this something that would be considered?

**Comment by davilima6** [comment]
I agree that'd be a powerful feature. In the meantime I'm considering working around with something like this: https://codepen.io/davilima6-the-lessful/pen/LYjMoNq
